homeassistant/components/husqvarna_automower/switch.py

```diff
@@ -32,12 +31,28 @@ async def async_setup_entry(
     """Set up switch platform."""
     coordinator = entry.runtime_data
     current_work_areas: dict[str, set[int]] = {}
-    current_stay_out_zones: dict[str, set[str]] = {}
 
-    async_add_entities(
-        AutomowerScheduleSwitchEntity(mower_id, coordinator)
-        for mower_id in coordinator.data
-    )
+    def _async_add_new_lock(mower_id: str) -> None:
+        async_add_entities([AutomowerScheduleSwitchEntity(mower_id, coordinator)])
+
+    def _async_add_new_stay_out_zones(mower_id: str, stay_out_zone_uid: str) -> None:
+        async_add_entities(
+            [StayOutZoneSwitchEntity(coordinator, mower_id, stay_out_zone_uid)]
+        )
+
+    # Add new lock entities for each mower
+    for mower_id in coordinator.data:
+        _async_add_new_lock(mower_id)
+
+    # Add new stay-out zone entities for each mower that has stay-out zones
+    for mower_id in coordinator.data:
+        mower_data = coordinator.data[mower_id]
+        if (
+            mower_data.capabilities.stay_out_zones
+            and mower_data.stay_out_zones is not None
+        ):
+            for stay_out_zone_uid in mower_data.stay_out_zones.zones:
+                _async_add_new_stay_out_zones(mower_id, stay_out_zone_uid)
```

Comment:
the main thing is you'll need to separate "normal" entity adding vs callback entity adding or you're calling `async_add_entities` here multiple times, which we want to avoid. So maybe also change your callback methods to accepting lists, so you can add new stay out zones together, if there are multiple. The next thing is you'll need to add the callbacks to a list in the coordinator from where they get called from the coordinator, where you also keep track of what you already added centrally. But as @joostlek said, it's gonna be different tuples for dynamic entities vs an entire dynamic device.

Reply:
Okay, I think I got it now.